src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -2783,6 +2774,14 @@ public void onSuccess(Void result) {
             // Go to entity's parent
             Place gotoPlace = createDeletePlace();
             getGlobalApplicationState().getPlaceChanger().goTo(gotoPlace);
+            KeyFactory keyFactory = keyFactoryProvider.getKeyFactory(
+              authenticationController.getCurrentUserAccessToken()
+            );
+            queryClient.invalidateQueries(
+              InvalidateQueryFilters.create(
+                keyFactory.getTrashCanItemsQueryKey()
+              )
+            );
```

Comment:
Use queryClient.invalidateQueries and InvalidateQueryFilters to invalidate the trashcan cache only when deletion is successful